pkg/utils/sort.go

```diff
@@ -80,25 +95,20 @@ func TopologicalSort(e []corev1.EnvVar) {
 	graph := make(map[int][]int)
 	n := len(e)
 
-	// initialize the dependency graph
+	// Initialize the dependency graph
 	for i := 0; i < n; i++ {
 		inDegree[i] = 0
 		graph[i] = make([]int, 0)
 	}
 
 	for i := 0; i < n; i++ {
 		for j := i; j < n; j++ {
-			// If a depends on b or b depends on a, then use their input order
-			if DependsOn(e[i], e[j]) || DependsOn(e[j], e[i]) {
```

Comment:
Please refer to https://github.com/risingwavelabs/risingwave-operator/pull/298, we have some discussion about `DependsOn` relation.

Reply:
Yep, but recent issues indicate that it's error-prone to do so.